### PR TITLE
Adds KOKKOS variant.

### PR DESCRIPTION
**Pull request by dipietrantonio**

### CODE REVIEW

**Comment by marcodelapierre** [comment]
Re: the Caliper recipe

I can confirm that building Caliper with Kokkos support does not require the latter as a dependency. 
If I remember correctly, the build option compiles an additional plugin, that can be used at runtime to profile Kokkos-based programs.

**Review by alalazo — COMMENTED** [review]
PR LGTM. The only question I have is if we double checked whether the new variant applies to all supported versions.

**Comment by daboehme** [comment]
LGTM. Kokkos support was added in Caliper v2.3.0 BTW, though no one should be using anything this old.

File: var/spack/repos/builtin/packages/caliper/package.py [hunk 1]
```diff
@@ -60,6 +60,7 @@ class Caliper(CMakePackage, CudaPackage, ROCmPackage):
     variant("sosflow", default=False, description="Enable SOSflow support")
     variant("fortran", default=False, description="Enable Fortran support")
     variant("variorum", default=False, description="Enable Variorum support")
+    variant("kokkos", default=True, description="Enable Kokkos profiling support")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/caliper/package.py]:
```suggestion
    variant("kokkos", default=True, when="@2.3.0:", description="Enable Kokkos profiling support")
```

Comment (alalazo) [reply var/spack/repos/builtin/packages/caliper/package.py]: > ... though no one should be using anything this old.

@daboehme Should we also deprecate versions? If so which ones?

Comment (daboehme) [reply var/spack/repos/builtin/packages/caliper/package.py]: > @daboehme Should we also deprecate versions? If so which ones?

Good question. Not sure what the spack policy on deprecating versions is, I thought they just keep everything forever. But certainly everything older than v2.8 can be deprecated at this point.

**Review by alalazo — APPROVED** [review]

File: var/spack/repos/builtin/packages/caliper/package.py (hunk 1 above)
Comment (alalazo) [reply var/spack/repos/builtin/packages/caliper/package.py]: @daboehme Thanks, I'll submit another PR for deprecating versions

**Pull request merged**